include/simics/dmllib.h

```diff
@@ -1976,8 +1976,8 @@ _register_port_legacy_proxy_attr(conf_class_t *devcls, conf_class_t *portcls,
                 is_bank ? "bank" : "port", portname, attrname);
         SIM_register_typed_attribute(
                 devcls, name,
-                getter ? _get_legacy_proxy_attr : NULL, data,
-                setter ? _set_legacy_proxy_attr : NULL, data,
+                getter ? _get_legacy_proxy_attr : NULL, getter ? data : NULL,
+                setter ? _set_legacy_proxy_attr : NULL, setter ? data : NULL,
```

Comment:
Does this make any observable difference? Or does the current state of affairs violate `SIM_register_type_attribute`'s API? If the answers to both are "no", then I must question why we should bother.

Reply:
To clarify: this the answer to both questions are "no", then this is an unnecessary complication. It'd be meaningless code, which is of negative value.

Reply:
@lwaern-intel AFAIK there is no observable difference. I came across this place when I decided to check if there are any attributes registered that have `NULL` getter or setter but non-`NULL` data pointer. My intention was to find possible errors where a `NULL` getter or setter was specified by mistake. I don't remember if any errors were found.
I think that passing NULL for data would be more consistent.
If you still think that this is an unnecessary complication, please just reject the pull request. No worries here.